Error out if multiple pipelineruns with same name are found

sm43 opened the pull request:
if there are multiple pipelineruns with sama name in .tekton directory then return an error.

Closes #1192 

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] ♽  Run `make test lint` before submitting a PR (ie: with [pre-commit](https://pipelinesascode.com/dev/tools), no need to waste CPU cycle on CI
- [ ] 📖 If you are adding a user facing feature or make a change of the behavior, please verify that you have documented it
- [ ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [ ] 🎁 If that's something that is possible to do please ensure to check if we can add a e2e test.
- [ ] 🔎 If there is a flakiness in the CI tests then don't *necessary* ignore it, better get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it. (token rate limitation may be a good reason to skip).

sm43 commented:
I am forgetting golang after looking at elixir 😄 so a little refresh!
need blessings for CI 🙇‍♂️ @chmouel

chmouel commented:
very cool!! 

we'll need to add you in the OWNER list

chmouel commented:
i think the e2e error is relevant, (can you see them)?

and we need somewhere in the doc mentioning this

sm43 commented:
Yeah I am guessing that we use the same pipelinerun template to create pr for the failing test. Due to which it is failing. Will take a look tomorrow. :)

chmouel commented:
/retest

chmouel commented on pkg/pipelineascode/match.go:
```diff
@@ -185,6 +185,11 @@ func (p *PacRun) getPipelineRunsFromRepo(ctx context.Context, repo *v1alpha1.Rep
 		return nil, nil
 	}
 
+	if err := pipelineRunsWithSameName(pipelineRuns); err != nil {
```
I think it may be better to get this inside those functions (by changing the logic to not exit too early):

https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/resolve/resolve.go#L102-L118

for pipeline and tasks, this would work as well for tkn pac resolve.

sm43 committed:
Error out if multiple pipelineruns with same name are found

if there are multiple pipelineruns with sama name in .tekton directory
then return an error.

sm43 commented:
/retest

chmouel commented:
i fixed the git.io thing, updated your pr and let see how it goes..

sm43 commented:
> `🔔 Shame 🔔`

😄 this is nice. will remind each time 🤣

chmouel commented:
haha i didn't wrote those strings, but it could have be me.... 😅 

it's still a bit buggy, it's go coverage being buggy generating the bug report...

sm43 commented:
> haha i didn't wrote those strings, but it could have be me.... 😅
> 
> it's still a bit buggy, it's go coverage being buggy generating the bug report...

lol I thought it was you! :D
Now CI should pass 🤞

sm43 commented:
the last e2e passed. @chmouel take a look, this should be good to go! https://github.com/openshift-pipelines/pipelines-as-code/actions/runs/4544257389

chmouel commented on test/pkg/payload/get_entries.go:
```diff
@@ -26,6 +28,8 @@ func GetEntries(yamlfile map[string]string, targetNS, targetBranch, targetEvent
 		name := strings.TrimSuffix(filepath.Base(target), filepath.Ext(target))
 		extraParams["PipelineName"] = name
+		// add some random character to name so that each PR has different name
+		extraParams["PipelineName"] = extraParams["PipelineName"] + "-" + strings.ToLower(random.AlphaString(4))
```
```suggestion
		extraParams["PipelineName"] = name + "-" + strings.ToLower(random.AlphaString(4))
```  

ans remove line 30,

chmouel commented:
/retest

chmouel commented:
congrats on your first contribution 😝

chmouel commented:
we maybe should have had this specified in the doc, i can add that or feel free to send a pr 🙇🏻‍♂️

sm43 commented:
> congrats on your first contribution 😝

Thank you! 😄 finally have a commit \o/
can you take care of the doc, i can create PR but not sure today or tomorrow

chmouel commented:
sure will do
